desc/magnetic_fields.py

```diff
@@ -1846,6 +1853,326 @@ def from_surface(
         )
 
 
+class OmnigenousField(Optimizable, IOAble):
+    """A magnetic field with perfect omnigenity (but is not necessarily analytic).
+
+    Notes
+    -----
+    Doesn't conform to MagneticField API, as it only knows about :math:`|B|` in
+    computational coordinates, not vector B in lab coordinates.
+
+    Parameters
+    ----------
+    L_B : int
+        Resolution of the radial Chebyshev polynomials for magnetic well parameters.
+    M_B : int
+        Number of monotonic spline knots per surface of the magnetic well parameters.
+    L_x : int
+        Resolution of the radial Chebyshev polynomials for the omnigenity parameters.
+    M_x : int
+        Resolution of the Fourier series in eta for the omnigenity parameters.
+    N_x : int
+        Resolution of the Fourier series in alpha for the omnigenity parameters.
+    NFP : int
+        Number of field periods.
+    helicity : tuple, optional
+        Type of pseudo-symmetry (M, N). Default = toroidal contours (1, 0).
+    B_lm : ndarray, optional
+        Magnetic well parameters describing |B|(ρ,η). These values are a flattened 2D
+        array of shape (L_B + 1, M_B), where the rows are Chebyshev coefficients
+        corresponding to the modes in `B_basis` for the radial variation, and the
+        columns are the values of |B| atlinearly spaced monotonic spline knots.
+        If not supplied, `B_lm` defaults to a constant field of 1 T.
```

Comment:
would an actual well be a better default?

Reply:
Not necessarily. OT/OH should have a mirror ratio of 0 on axis (ie, no well); it's only OP that has a well on axis. And if we use an actual well instead then we also need to make a choice about it's depth, slope, etc. Realistically users should give their own initial guess anyways if this simple default doesn't work for them.